apps/ejabberd/src/ejabberd_c2s.erl

```diff
@@ -1890,44 +1888,52 @@ presence_update_to_available(Acc, StateData, From, Packet) ->
 
     FromUnavail = (StateData#state.pres_last == undefined) or StateData#state.pres_invis,
     ?DEBUG("from unavail = ~p~n", [FromUnavail]),
-    case FromUnavail of
-        true ->
-            Acc2 = ejabberd_hooks:run_fold(user_available_hook,
-                                           NewStateData#state.server,
-                                           Acc1,
-                                           [NewStateData#state.jid]),
-            NewStateData1
-            = case NewPriority >= 0 of
-                  true ->
-                      {_, _, Pending} = ejabberd_hooks:run_fold(
-                                          roster_get_subscription_lists,
-                                          NewStateData#state.server,
-                                          {[], [], []},
-                                          [StateData#state.user, NewStateData#state.server]),
-                      % TERMINATE - those two functions will be rewritten in the next stage
-                      resend_offline_messages(NewStateData),
-                      resend_subscription_requests(NewStateData#state{
-                                                     pending_invitations = Pending});
-                  false ->
-                      NewStateData
+    presence_update_to_available(Acc1, FromUnavail, OldPriority, NewPriority, From,
+                                 Packet, NewStateData).
+
+%% @doc the first one is run when presence changes from unavailable to anything else
+-spec presence_update_to_available(Acc :: mongoose_acc:t(),
+                                   FromUnavailable :: boolean(),
+                                   OldPriority :: integer(),
+                                   NewPriority :: integer(),
+                                   From :: ejabberd:jid(),
+                                   Packet :: exml:element(),
+                                   StateData :: state()) -> {mongoose_acc:t(), state()}.
+presence_update_to_available(Acc, true, _, NewPriority, From, Packet, StateData) ->
```

Comment:
IMO `(true, Acc, ..., State)` but it all depends on the use case :) In general, I personally think that if we pattern match on a parameter in a function head, it's more readable if this is the first parameter.